tiatoolbox/utils/misc.py

```diff
@@ -1217,19 +1217,19 @@ def dict_to_store(
     # get relevant keys
     class_probs = patch_output.get("probabilities", [])
     preds = patch_output.get("predictions", [])
+
     patch_coords = np.array(patch_output.get("coordinates", []))
     if not np.all(np.array(scale_factor) == 1):
         patch_coords = patch_coords * (np.tile(scale_factor, 2))  # to baseline mpp
     labels = patch_output.get("labels", [])
     # get classes to consider
-    if len(class_probs) == 0:
-        classes_predicted = np.unique(preds).tolist()
-    else:
-        classes_predicted = range(len(class_probs[0]))
+    classes_predicted = np.unique(preds).tolist()
```

Comment:
Requesting @measty's review on this change:

While testing AnnotationStore for the WSI output, I used the existing `mini_wsi_svs = Path(sample_wsi_dict["wsi2_4k_4k_svs"])` image as input to the updated Engine ABC's `run()` method. 

The Post process step now has an extra parameter, `str: output_type` which can be "AnnotationStore", which calls the `utils.misc.dict_to_store()` method, which throws a random `Key Error` every time. Upon debugging, the `classes_predicted = np.unique(preds).tolist()` resolves it as in this given case, the classes_predicted is not necessarily all the numbers in 0 to 9. Please advise if we should remove that if check, or if this is expected behaviour. 

```============================================================================================== PASSES ===============================================================================
_______________________________________________________________________________ test_engine_run_wsi_annotation_store ________________________________________________________________________________
--------------------------------------------------------------------------------------- Captured stdout call ----------------------------------------------------------------------------------------
|2023-11-27|13:34:26.736| [INFO] When providing multiple whole slide images, the outputs will be saved, and the locations of outputs will be returned to the calling function.
[DEBUG] classes_predicted  [0, 1, 2, 3, 4, 6, 7]
[DEBUG] class_probs shape  (69, 9)
[DEBUG] preds shape  (69,)
[DEBUG] preds  [7, 3, 2, 3, 3, 3, 3, 4, 4, 3, 3, 3, 3, 7, 3, 3, 6, 4, 3, 4, 3, 3, 3, 4, 6, 0, 0, 0, 0, 3, 0, 0, 3, 3, 3, 0, 2, 3, 3, 3, 1, 3, 6, 2, 3, 3, 3, 3, 3, 0, 4, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 7, 3, 3, 3, 3, 3, 3, 2]
[DEBUG] labels  []
[Debug 2] Labels  None
[DEBUG] class_dict  {0: 0, 1: 1, 2: 2, 3: 3, 4: 4, 6: 6, 7: 7}
[DEBUG] class_dict shape  ()```

Reply:
Hi, 
The classes_predicted cannot be changed that way, because when there are probabilities output, there will be predicted probabilities even for classes that don't appear in preds, and these will need to be saved.

The problem is rather in this part:
```
    if class_dict is None:
        # if no class dict create a default one
        class_dict = {i: i for i in np.unique(preds + labels).tolist()}
```
which needs to be modified to take into account the possibility that some classes for which a probability is output may not have an instance where that class 'wins' resulting in an instance classified as that class in preds.
Hopefully, the following modification would solve the issue:
```
    if class_dict is None:
        # if no class dict create a default one
        if len(class_probs) == 0:
              class_dict = {i: i for i in np.unique(preds + labels).tolist()}
        else:
              class_dict = {i: i for i in range(len(class_probs))}
```

Reply:
Hey Mark. Thank you, it is resolved now.